app/controllers/articles_controller.rb

```diff
@@ -1,28 +1,40 @@
 class ArticlesController < ApplicationController
   def index
-
+    @articles = Article.all
   end
   def show
     @article = Article.find(params[:id])
   end
   def new
-
+    @article = Article.new
   end
   def edit
-
+    @article = Article.find(params[:id])
   end
   def create
     #render method taking a hash as parameter
     #render plain: params[:article].inspect
     @article = Article.new(article_params)
-    @article.save
-    redirect_to @article
+    if @article.save
+      redirect_to @article
+    else
+      render 'new'
+    end
   end
   def update
-
+    @article = Article.find(params[:id])
+    if @article.update(article_params)
+      redirect_to @article
+    else
+      render 'edit'
+    end
   end
   def destroy
-
+    @article = Article.find(params[:id])
+    @article.destroy
+    #redirect_to article_path
+    #Still rendering the incorrect view. Fix it!
+    render 'new'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.